lib/guardian.ex

```diff
@@ -718,7 +718,7 @@ defmodule Guardian do
     case result do
       {:ok, _} -> result
       {:error, _} -> result
-      resp -> {:error, "Invalid return for #{mod}##{func} - #{inspect(resp)}"}
+      resp -> raise "Expected `{:ok, result}` or `{:error, reason}` from #{mod}##{func}, got #{inspect(resp)}"
```

Comment:
Incorporated @michalmuskala's feedback and tweaked the message to inform the user what the function expects.  Thoughts?